pkg/argocd/git.go

```diff
@@ -164,7 +195,7 @@ func commitChangesGit(app *v1alpha1.Application, wbc *WriteBackConfig, changeLis
 	// config, or taken from the application spec's targetRevision. If the
 	// target revision is set to the special value HEAD, or is the empty
 	// string, we'll try to resolve it to a branch name.
-	checkOutBranch := getApplicationSource(app).TargetRevision
+	checkOutBranch := getWriteBackBranch(app)
 	if wbc.GitBranch != "" {
 		checkOutBranch = wbc.GitBranch
 	}
```

Comment:
if the gitbranch is set, we always take it. So I think we only need to call the func `getWriteBackBranch` if wbc.GitBranch is empty (in the else clause).